technologies/FlexBE/README.md

```diff
@@ -55,10 +60,68 @@ To run start the following in separate terminals
+
+Specifically we demonstrate:
+* `LogState` - print message to terminal (both console and UI)
+* `LogKeyState` - print message containing user data value at specified key
+* `OperatorDecisionState` - allow operator to select among specified outcomes; in full autonomy the "suggested" outcome is selected.
+* `SelectionState` - uses `input_action_server` to allow user to select among given data
+* `InputState` - uses `input_action_server` to allow user to input simple data such as string, number, or list of numbers
+
+The `pyrobosim_flexbe_states` package under `technologies/FlexBE` includes:
+* `check_door_state.py` - Uses pyrobosim `RequestWorldState` service to check door status using non-blocking call
+* `detect_local_objects_state.py` - Invokes a pyrobosim `ExecuteTaskAction` to `detect` objects, then uses `RequestWorldState`
+* `detect_objects_state.py` - Invokes the pyrobosim `DetectObjects` action to detect objects, then adds to user data
+* `door_action_state.py` - Invoke `ExecuteTaskAction` to `open` or `close` an openable (including doors, dumpster, pantry, fridge, ...)
+* `follow_path_state.py` - Follow current path passed as user data into state
+* `monitor_battery_state.py` - Monitor battery state and return outcome if either low or high level detected (blocks, use in concurrent state)
+* `navigate_action_state.py` - Navigation action that combines planning and following
+* `next_room_state.py` - Does simple planning to determine adjacent room based on connected rooms
+* `pick_action_state.py` - Does pick action at current location
+* `place_action_state.py` - Does place action at current location
+* `plan_path_state.py` - Request plan from current location to target location
+
+The above states have extra logging information that is shown in the onboard behavior terminal.  Each state transition `on_start`, `on_enter`, `on_exit`, `on_pause`, `on_resume`, and `on_stop` are logged.  This is not recommended in regular states, but is done here for educational purposes.
+
+### Basic Demonstration Behaviors
+
+In order of increasing complexity
+
+Use `clear; ros2 run delib_ws_worlds run --ros-args -p problem_number:=1`
+
+* `test navigate` - Test navigation state
+* `Test Plan Path` - Simple plan then follow behavior with userdata
+* `test pick place` - Navigation with pick and place using behavior parameters
+* `DetectSelect` - Move, detect objects, select using `input_action_server`, move and place
+* `Go Beh` - Go to target location
+
+Use `clear; ros2 run delib_ws_worlds run --ros-args -p problem_number:=1`
+
+* `delib_ws_p2` - use behavior parameters plan to door, open door, then use `DetectSelect` embedded behavior
+* `delib_ws_p2_sm` - use hierarchical state machine to select door, open, then use `DetectSelect` embedded behavior
+* `Through Door` - Travel through doorway opening if necessary
+* `Traverse` - Go to specified target location, opening intermediate doors if necessary using `Go Beh` and `Through Door`
```

Comment:
is `Go Beh` correct, or a typo?

(I checked, and I guess it's correct -- maybe consider expanding out the name? Is it supposed to mean "behavior"?)

Reply:
Was short for "Go Behavior"   I had some not well thought out names during testing that I should clean up, but I wanted to get something up.